From 0be09792a0670d0758fbee20659186faa2c0141e Mon Sep 17 00:00:00 2001
From: Gilles Sadowski <erans@apache.org>
Date: Tue, 18 Dec 2012 18:02:52 +0000
Subject: [PATCH] MATH-922 Parent class's method "getCovariances()" assumes
 that the protected field "point" contains the result of the optimization.

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/math/trunk@1423555 13f79535-47bb-0310-9956-ffa450edef68
---
 .../optimization/general/LevenbergMarquardtOptimizer.java   | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/main/java/org/apache/commons/math3/optimization/general/LevenbergMarquardtOptimizer.java b/src/main/java/org/apache/commons/math3/optimization/general/LevenbergMarquardtOptimizer.java
index 6ce1f1a390..0c41d3eb9d 100644
--- a/src/main/java/org/apache/commons/math3/optimization/general/LevenbergMarquardtOptimizer.java
+++ b/src/main/java/org/apache/commons/math3/optimization/general/LevenbergMarquardtOptimizer.java
@@ -374,6 +374,8 @@ protected PointVectorValuePair doOptimize() {
             if (maxCosine <= orthoTolerance) {
                 // Convergence has been reached.
                 setCost(currentCost);
+                // Update (deprecated) "point" field.
+                point = current.getPoint();
                 return current;
             }
 
@@ -482,6 +484,8 @@ protected PointVectorValuePair doOptimize() {
                         // we use the vectorial convergence checker
                         if (checker.converged(iter, previous, current)) {
                             setCost(currentCost);
+                            // Update (deprecated) "point" field.
+                            point = current.getPoint();
                             return current;
                         }
                     }
@@ -508,6 +512,8 @@ protected PointVectorValuePair doOptimize() {
                      ratio <= 2.0) ||
                     delta <= parRelativeTolerance * xNorm) {
                     setCost(currentCost);
+                    // Update (deprecated) "point" field.
+                    point = current.getPoint();
                     return current;
                 }
 
